artichoke-backend/src/extn/core/time/offset.rs

```diff
@@ -0,0 +1,78 @@
+use crate::extn::core::time::Offset;
+use crate::extn::prelude::*;
+use crate::convert::{implicitly_convert_to_int, implicitly_convert_to_string};
+use crate::extn::core::symbol::Symbol;
+
+const MAX_FLOAT_OFFSET: f64 = i32::MAX as f64;
+const MIN_FLOAT_OFFSET: f64 = i32::MIN as f64;
+
+impl TryConvertMut<Value, Option<Offset>> for Artichoke {
+    type Error = Error;
+
+    fn try_convert_mut(&mut self, options: Value) -> Result<Option<Offset>, Self::Error> {
+        let hash: Vec<(Value, Value)> = self.try_convert_mut(options)?;
+
+        // Short circuit the offset parameter fetching.
+        if hash.len() == 0 {
+            return Ok(None)
+        }
+
+        // Extract the value from the options hash with the key `:in`,
+        // rejecting any other values as ArgumentError.
+        let mut in_param: Option<Value> = None;
+
+        for (mut key, value) in hash.iter() {
+            let k = unsafe { Symbol::unbox_from_value(&mut key, self)? }.bytes(self);
+            if k == b"in" {
+                in_param = Some(*value)
+            } else {
+                let mut message = b"unknown keyword: ".to_vec();
+                message.extend(k.to_vec());
+                Err(ArgumentError::from(message))?
+            }
+        }
+
+        if let Some(mut in_param) = in_param {
```

Comment:
to avoid some rightward drift here we can rewrite this as:

```suggestion
        let mut in_param = in_param.unwrap_or_else(|| unreachable!("..."));
```

Reply:
or `in_param.expect("...")`

Reply:
Good catch!